tlair/src/main/scala/at/forsyte/apalache/tla/lir/TlaLevel.scala

```diff
@@ -0,0 +1,92 @@
+package at.forsyte.apalache.tla.lir
+
+/**
+ * The level of a TLA+ expression, as explained in: Lamport. Specifying Systems, p. 322.
+ */
+sealed trait TlaLevel extends Ordered[TlaLevel] {
+
+  /**
+   * The level number as assigned in the book.
+   */
+  val level: Int
+
+  /**
+   * A textual representation of the level.
+   */
+  val name: String
+
+  /**
+   * Compare this level to another level.
+   *
+   * @param that the level to compare
+   * @return 0, if the levels are equal; a negative number if `this` is smaller than `that`, and a positive number
+   *         if `this` is larger than `that`.
+   */
+  override def compare(that: TlaLevel): Int = {
+    level - that.level
+  }
+
+  /**
+   * Join the levels by computing the smallest level that covers both of `this` and `that`.
```

Comment:
I'm surprised that joining gives the least, rather than greatest level. Is this because its actually more difficult to handle , e.g., constants than handling temporal formulas?

Reply:
no-no, it's actually the maximum of the two levels, just badly formulated